### PR TITLE
Add keycloak_default_groups resource for creating realm default groups

### LINKED ISSUE #144: Add support for default groups
AFAICT, there is no way to specify default groups at the moment. This would be fairly easy to implement, since it's a really basic resource. The relevant endpoints are:

    GET /{realm}/default-groups (returns a list of all default groups)
    PUT /{realm}/default-groups/{groupId} (add a group to the list of default groups)
    DELETE /{realm}/default-groups/{groupId} (delete a group from the list of default groups)

The first way that comes to mind for how this could be implemented is having a very basic resource for each default group that simply maps a realm id to a group id. This maps very well to the keycloak API. The HCL would look something look this:

```
resource "keycloak_group" "example_group" {
  realm_id = "${keycloak_realm.realm.id}"
  name     = "example-group"
}

resource "keycloak_default_group" "example_group" {
  realm_id = "${keycloak_realm.realm.id}"
  group_id = "${keycloak_group.example_group.id}"
}
```

We could also have a default groups property on realms, or have a default groups resource, or have a default group property on groups, but I don't think those solutions map very well to the API. Additionally, we could specify it on the realm, but then we get into a scenario where if a realm references groups that need to be created after the realm is then there's a potential deadlock.

I would be happy to take a crack at adding this feature.

### LINKED ISSUE #145: Errors aren't checked properly when removing a group membership user from a group
https://github.com/mrparkers/terraform-provider-keycloak/blob/0f1fe1bad1e107f0199bb93466b18003450e7a79/provider/resource_keycloak_group_memberships.go#L106

I think this code should be returning `err` rather than `nil`. As-is, if there's an error removing a user from a group nothing will happen.

### PR DESCRIPTION
**Pull request by fharding1**
## Summary

This PR adds a new `keycloak_default_groups` resource which allows you to manage default groups for a realm.

Closes #144 
Closes #145 

## TODO

- [x] Validate that a group membership resource doesn't exist or figure out some other solution for that (or just document that using a default group and a group membership resource will lead to funky results), see https://github.com/mrparkers/terraform-provider-keycloak/issues/144#issuecomment-526634858
- [x] Add tests
- [x] Add ability to import default groups
- [x] Add documentation
- [x] Add an example

### CODE REVIEW

**Comment by fharding1** [comment]
I changed the example to be formatted with the canonical `terraform fmt`. If this isn't desired, I can revert it.

**Comment by fharding1** [comment]
Hmm, tests are failing on `TestAccKeycloakDataSourceOpenidClient_basic` and `TestAccKeycloakDataSourceOpenidClientAuthorizationPolicy_basic`, but I didn't touch those.

File: provider/resource_keycloak_default_groups_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	var defaultGroupResources []string
+	for _, defaultGroup := range defaultGroups {
+		defaultGroupResources = append(defaultGroupResources, fmt.Sprintf(`"${keycloak_group.%s.id}"`, defaultGroup))
+	}
+
+	out += fmt.Sprintf(`
+resource "keycloak_default_groups" "group_default" {
+	realm_id = "${keycloak_realm.realm.id}"
+	group_ids = [%s]
```
Comment (mrparkers) [inline provider/resource_keycloak_default_groups_test.go]: I have a helper function in the test package called `arrayOfStringsForTerraformResource` that does this for you, could you refactor this to use that function?

Comment (fharding1) [reply provider/resource_keycloak_default_groups_test.go]: Perfect! Can do.

**Comment by fharding1** [comment]
Had to force push to get rid of an accidentally commited update to the `go.mod` file (thanks go1.13rc2 :smile: )

**Comment by fharding1** [comment]
Thanks for the review and for merging! :tada:

**Pull request merged**